BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/CredentialEnvelope.cs

```diff
@@ -16,13 +16,19 @@ public abstract class CredentialEnvelope
         /// <summary>
         /// Gets an Azure.Core.AccessToken.
         /// </summary>
+        /// <remarks>
+        /// Whomever uses this MUST verify that it's called within <see cref="SdkInternalOperationsMonitor.Enter"/> otherwise dependency calls will be tracked.
```

Comment:
add a TODO to decide if we want to call SdkIntern.Enter inside the implementation itself.

Reply:
done. 
also tracking remaining todo's here: #2190